### ISSUE TITLE
Spotify Changed Variables?

### ISSUE THREAD

**Issue by ajoshuasmith** [carharv/SpotifyHistoryJSONtoCSV#1]
Apologies as I am not the savviest with coding languages.

I have been trying to get this going with a new export from Spotify 4/12/22 and it seems that they changed the naming? I only assume this because when I am trying to run this script I am getting some errors 

`AttributeError: 'DataFrame' object has no attribute 'ms_played'`
`AttributeError: 'DataFrame' object has no attribute 'master_metadata_album_artist_name'`

Example of the StreamingHistoryX.json

`{
    "endTime" : "2021-04-11 13:00",
    "artistName" : "Trippie Redd",
    "trackName" : "Taking A Walk",
    "msPlayed" : 121680
  },`

Since there is no albumName will this import still work since the notes say we do want ArtistName,Album,TrackName,ect?

The other .json files Spotify provided
![image](https://user-images.githubusercontent.com/78829059/163053048-da5be6ed-1134-4a7c-bcad-c7f544d32955.png)

Edit ----

I am realizing that they did not send me the full account history.. xD

**Comment by carharv** [comment]
Yep so there are two types of history you can request. Just the past year's history, which is what it looks like you have above, and your lifetime history. 

With that being said, you can easily modify the script to work with the data you have. You'll want to look at lines 44-50 if you're using the .ipynb.

|Instead Of|You Would Use|
| --- |       --- |
|`totalMsListened += jsonCombinedDf.ms_played[x]`| `totalMsListened += jsonCombinedDf.msPlayed[x]`|
|`if jsonCombinedDf.ms_played[x] >= (minPlayTime*1000)`|`if jsonCombinedDf.msPlayed[x] >= (minPlayTime*1000)`|
|`trackArtist = jsonCombinedDf.master_metadata_album_artist_name[x]`|`trackArtist = jsonCombinedDf.artistName[x]`|

There are a few more variable names that it seems like you will need to change to match the JSON format of your file. Let me know if you still can't get it going with those changes and I'll work on adding a script to work with the partial history files.
